llarp/service/endpoint.cpp

```diff
@@ -805,6 +808,30 @@ namespace llarp
       return {{"LOKINET_ADDR", m_Identity.pub.Addr().ToString()}};
     }
 
+    void
+    Endpoint::FlushRecvData()
+    {
+      do
+      {
+        auto maybe = m_RecvQueue.tryPopFront();
+        if(not maybe.has_value())
+          return;
+        auto ev = std::move(maybe.value());
+        ProtocolMessage::ProcessAsync(ev.fromPath, ev.pathid, ev.msg);
+      } while(true);
+    }
```

Comment:
not a big fan of `do { ... } while`, esp. `while(true)`... just sayin'

Reply:
It feels verbose/redundant to do the assignment twice (once before the loop, once at the end of each iteration), but I agree that it would be more clear to write the loop as `while (maybe.has_value())`.